subprojects/base-services/src/main/java/org/gradle/internal/operations/BuildOperation.java

```diff
@@ -16,12 +16,14 @@
 
 package org.gradle.internal.operations;
 
+import org.gradle.internal.progress.BuildOperationDescriptor;
+
 /**
  * Represents some chunk of work.
  */
 public interface BuildOperation {
     /**
      * Returns a human consumable name for this operation.
      */
-    String getDescription();
+    BuildOperationDescriptor.Builder description();
```

Comment:
Can you elaborate some more what direction you propose? One alternative that Jendrik and I discussed was having individual methods like `getShortDescription`, `getDescription`, `getProgressDescription` etc. That would make it easy to do `BuildOperation extends Describable`. We'd need a convenience base class for the very common case that most of these fields are null though.